R/input_checking_functions.R

```diff
@@ -304,27 +349,46 @@ check_time_is_sequential <- function(
     }
 
     timestep <- parameters[['timestep']]
+    first_diff <- diff(time, differences = 1)
+    second_diff <- diff(time, differences = 2)
+    is_linear <- abs(second_diff) < rtol
+    is_spaced_by_timestep <- abs(first_diff - timestep) < rtol
+    
+    if (!all(is_linear)) {
+        idx <-which(!is_linear)
+        msg <- paste0(
+            c("`time` not linear.", "The `time` variable in the drivers is not linear and not spaced by `timestep`. Drivers may contain gaps. Drivers should look like:",
+            "`drivers$time <- seq(1, nrow(drivers), timestep)`", 
+            "+ Check `timestep` matches `diff(time)`", 
+            "+ Check rows:", paste(head(idx), collapse=', ') ), collapse='\n   ')
+        return(msg)
+    }
 
-    if (!is_evenly_spaced(time, timestep, rtol)) {
-        return("The `time` variable is not spaced by `timestep`.")
+    if (!all(is_spaced_by_timestep)) {
+        idx <-which(!is_spaced_by_timestep)
+        msg <- paste0(
+            c("Timestep mismatch.", "The `time` variable in the drivers is linear but not spaced by `timestep`. Drivers may contain gaps. Drivers should look like:",
+            "`drivers$time <- seq(1, nrow(drivers), timestep)`", 
```

Comment:
The comment that "Drivers should look like: `drivers$time <- seq(1, nrow(drivers), timestep)`" seems a little odd. People are not usually specifying time this way. I would avoid putting R code in the message since people may think they are supposed to run it. It's probably enough to say that `time` is not spaced by `timestep`.